src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1665,6 +1782,27 @@ UndistributeTable(Oid relationId)
 							NULL, None_Receiver, NULL);
 	}
```

Comment:
In case of alter_table_set_access_method we should append one more command to preserve the distributedness of the table (e.g. `SELECT create_reference_table(...)`, `SELECT create_citus_local_table(...)`, `SELECT create_distributed_table(..., colocate_with := 'original_table')`) before doing the INSERT..SELECT. 

It probably makes sense to write a new function that returns that command for a given relation ID or NULL / empty list for regular tables.

Reply:
RelayEventExtendNames and CitusProcessUtility do not really work with these queries. So I think I should call the CreateDistributedTable and CreateCitusLocalTable functions directly.